viabel/optimization.py

```diff
@@ -185,77 +198,88 @@ def descent_direction(self, grad):
         return descent_dir
 
 
-class WindowedAdagrad(StochasticGradientOptimizer):
-    """Adam optimization method
+class AveragedRMSProp(StochasticGradientOptimizer):
+    """Averaged RMSProp optimization method (Mukkamala and Hein, 2017, §4)
     
-    Parameters
-    -----------
-    window size : `int` optional
-        Window size used to store the square of the gradients. The default is 10
-    jitter: `float` optional
-        Small value used for numerical stability. The default is 1e-8
-
-    Returns
-    ----------
-    descent_dir : `numpy.ndarray`, shape(var_param_dim,)
-        Descent direction of the optimization algorithm
-    """
-
-    def __init__(self, learning_rate, *, weight_decay=0, window_size=10, jitter=1e-8,
-                 diagnostics=False):
-        self._window_size = window_size
-        self._jitter = jitter
-        super().__init__(learning_rate, weight_decay=weight_decay, diagnostics=diagnostics)
-
-    def reset_state(self):
-        """
-        resetting the state
-        """
-        self._history = []
-
-    def descent_direction(self, grad):
-        self._history.append(grad**2)
-        if len(self._history) > self._window_size:
-            self._history.pop(0)
-        mean_grad_squared = np.mean(self._history, axis=0)
-        descent_dir = grad / np.sqrt(self._jitter + mean_grad_squared)
-        return descent_dir
-
-
-class Adagrad(StochasticGradientOptimizer):
-    """Adagrad optimization method
+    Uses averaged squared gradient by setting :math:`\\beta_k = 1-1/k` such that 
+    
+    .. math::
+        \\nu^{(k+1)} = \\beta_k \\nu^{(k)} + (1-\\beta_k) \\hat{g}^{(k)} \\cdot  \\hat{g}^{(k)}.
+    
+    Then,
+    
+    .. math::
+        \\nu^{(k+1)} = (k+1)^{-1} \\sum^k_{k^\\prime =0}\\hat{g}^{(k)} \\cdot  \\hat{g}^{(k)},
+    
+    where :math:`\\nu^{(k)}` converges to a constant almost surely under certain 
+    conditions.
     
     Parameters
     -----------
     jitter: `float` optional
         Small value used for numerical stability. The default is 1e-8
+    component_wise: `boolean` optional
+        Indication of  component wise discent direction computation
```

Comment:
Not clear. Do you mean "indicator of whether to do component-wise normalization of descent direction"?